llmstudio/engine/providers/azure.py

```diff
@@ -44,48 +56,284 @@ async def generate_client(
         self, request: AzureRequest
     ) -> Coroutine[Any, Any, Generator]:
         """Generate an AzureOpenAI client"""
+
+        self.is_llama = "llama" in request.model.lower()
+        self.has_tools = request.tools is not None
+
         try:
             if request.base_url or self.BASE_URL:
                 client = OpenAI(
                     api_key=request.api_key or self.API_KEY,
                     base_url=request.base_url or self.BASE_URL,
                 )
-                return await asyncio.to_thread(
-                    client.chat.completions.create,
-                    model=request.model,
-                    messages=(
-                        [{"role": "user", "content": request.chat_input}]
-                        if isinstance(request.chat_input, str)
-                        else request.chat_input
-                    ),
-                    stream=True,
-                )
             else:
                 client = AzureOpenAI(
                     api_key=request.api_key or self.API_KEY,
                     azure_endpoint=request.api_endpoint or self.API_ENDPOINT,
                     api_version=request.api_version or self.API_VERSION,
                 )
-                return await asyncio.to_thread(
-                    client.chat.completions.create,
-                    model=request.model,
-                    messages=(
-                        [{"role": "user", "content": request.chat_input}]
-                        if isinstance(request.chat_input, str)
-                        else request.chat_input
-                    ),
-                    functions=request.functions,
```

Comment:
should we keep both "functions" and "tools" logging an deprecation warning for those using functions?

Reply:
I think this would be a good approach to the problem.